### ISSUE TITLE
[bug] Disabling remote prevents build from proceeding

### ISSUE THREAD

**Issue by cassava** [conan-io/conan#9183]
Disabling a broken remote prevents me from building a Conan package, even though I have all packages I need in my local cache.

### Environment Details (include every applicable attribute)
  * Operating System: Ubuntu 18.04
  * Compiler: GCC=8
  * Conan version: 1.37.2
  * Python version: 3.6.9

### Steps to reproduce

1. `conan remote disable artifactory`
2. `conan install --build=missing --install-folder=build -g cmake`

### Logs

1. Timeout
```console
$ conan install --build=missing --install-folder build -g cmake .
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=RelWithDebInfo
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

ERROR: HTTPSConnectionPool(host='artifactory.example.com', port=443): Max retries exceeded with url: /artifactory/api/conan/conan-local/v1/conans/search?q=fmt%2F%2A&ignorecase=False (Caused by ReadTimeoutError("HTTPSConnectionPool(host='artifactory.example.com', port=443): Read timed out. (read timeout=60.0)",))

Unable to connect to artifactory=https://artifactory.example.com/artifactory/api/conan/conan-local
1. Make sure the remote is reachable or,
2. Disable it by using conan remote disable,
Then try again.
```

2. Disable artifactory remote:
```console
$ conan remote disable artifactory
```

3. Try building again:
```console
$ conan install --build=missing --install-folder build -g cmake .
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=RelWithDebInfo
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

ERROR: Remote 'artifactory' is disabled
```

**Comment by memsharded** [comment]
Hi @cassava 

Thanks for the detailed report, already reproduced, indeed seems a bug, and I am proposing some potential fix in #9184

**Comment by memsharded** [comment]
#9184 has been last minuted merged into 1.38, so it will be released today
